static/app/components/profiling/flamegraph.tsx

```diff
@@ -24,44 +28,38 @@ function getTransactionConfigSpace(profiles: Profile[]): Rect {
   return new Rect(startedAt, 0, endedAt - startedAt, 0);
 }
 interface FlamegraphProps {
+  onImport: ProfileDragDropImportProps['onImport'];
   profiles: ProfileGroup;
 }
 
 function Flamegraph(props: FlamegraphProps): ReactElement {
   const flamegraphTheme = useFlamegraphTheme();
-  const [{sorting, view, synchronizeXAxisWithTransaction}, dispatch] =
-    useFlamegraphPreferences();
-  const canvasPoolManager = useMemo(() => new CanvasPoolManager(), []);
-
-  const [activeProfileIndex, setActiveProfileIndex] = useState<number | null>(null);
-  const [importedProfiles, setImportedProfiles] = useState<ProfileGroup | null>(null);
+  const [{sorting, view, xAxis}, dispatch] = useFlamegraphPreferences();
+  const {activeProfileIndex} = useFlamegraphProfilesValue();
 
-  // once an import occurs, it will always take precedence over the profile in the props
-  const profiles = importedProfiles ?? props.profiles;
+  const canvasPoolManager = useMemo(() => new CanvasPoolManager(), []);
 
   const flamegraph = useMemo(() => {
-    // if the activeProfileIndex is null, use the activeProfileIndex from the profile group
-    const profileIndex = activeProfileIndex ?? profiles.activeProfileIndex;
+    if (
+      typeof activeProfileIndex !== 'number' ||
+      props.profiles.profiles[activeProfileIndex] === undefined
+    ) {
+      return FlamegraphModel.Empty();
+    }
```

Comment:
I don't think you need an empty flame graph here. When the active profile index from the state is `null`, you should be using `props.profiles.activeProfileIndex`. This should also simplify a lot of the code below where you handle this empty state.

Reply:
I dont think this makes any difference to the end user tbh. What I wanted to avoid was reading from two different sources to avoid confusion. The correct way to handle this would be to have a "thread not found" screen instead of the empty view, but I want to tackle that separately